test/test_SW_VegProd.cc

```diff
@@ -159,4 +159,604 @@ namespace {
+
+        // Array holding all values from the estimation
+        double RelAbundanceL0[8]; // 8 = Number of types
+
+        // Array holding all values from estimation minus grasses
+        double RelAbundanceL1[5]; // 5 = Number of types minus grasses
+
+        double SumGrassesFraction = SW_MISSING;
+        double C4Variables[3];
+
+        Bool fillEmptyWithBareGround = swTRUE;
+        Bool warnExtrapolation = swTRUE;
+        Bool inNorthHem = swTRUE;
+
+        int deallocate = 0;
+        int allocate = 1;
+        int index;
+
+        // RelAbundanceL0 and inputValues indices
+        int succIndex = 0;
+        int forbIndex = 1;
+        int C3Index = 2;
+        int C4Index = 3;
+        int grassAnn = 4;
+        int shrubIndex = 5;
+        int treeIndex = 6;
+        int bareGround = 7;
+
+        // RelAbundanceL1 indices
+        int treeIndexL1 = 0;
+        int forbIndexL1 = 1;
+        int shrubIndexL1 = 2;
+        int grassesIndexL1 = 3;
+        int bareGroundL1 = 4;
+
+        double RelAbundanceL0Expected[8] = {0.0, 0.2608391, 0.4307062,
+                                                0.0, 0.0, 0.3084547, 0.0, 0.0};
+        double RelAbundanceL1Expected[5] = {0.0, 0.3084547, 0.2608391, 0.4307062, 0.0};
+
+        // Reset "SW_Weather.allHist"
+        SW_WTH_read();
+
+        // Allocate arrays needed for `calcSiteClimate()` and `averageClimateAcrossYears()`
+        allocDeallocClimateStructs(allocate, 31, &climateOutput, &climateAverages);
+
+        // Calculate climate of the site and add results to "climateOutput"
+        calcSiteClimate(SW_Weather.allHist, 31, 1980, inNorthHem, &climateOutput);
+
+        // Average values from "climateOutput" and put them in "climateAverages"
+        averageClimateAcrossYears(&climateOutput, 31, &climateAverages);
+
+        // Set C4 results, standard deviations are not needed for estimating vegetation
+        C4Variables[0] = climateAverages.minTemp7thMon_C;
+        C4Variables[1] = climateAverages.ddAbove65F_degday;
+        C4Variables[2] = climateAverages.frostFree_days;
+
+        // Estimate vegetation based off calculated variables and "inputValues"
+        estimatePotNatVegComposition(climateAverages.meanTemp_C, climateAverages.PPT_cm,
+            climateAverages.meanTempMon_C, climateAverages.PPTMon_cm, inputValues, shrubLimit,
+            SumGrassesFraction, C4Variables, fillEmptyWithBareGround, inNorthHem, warnExtrapolation,
+            grassOutput, RelAbundanceL0, RelAbundanceL1);
+
+        /*  ===============================================================
+                     Test when all input values are "SW_MISSING"
+            ===============================================================  */
+
+        /* Expect identical output to rSOILWAT2 (e.g., v5.3.1)
+         * NOTE: Command uses deprecated estimate_PotNatVeg_composition (rSOILWAT >= v.6.0.0)
+         ```{r}
+           rSOILWAT2::estimate_PotNatVeg_composition_old(

```

Comment:
- [ ] `estimate_PotNatVeg_composition_old()` is not an exported function. This must be instead `rSOILWAT2:::estimate_PotNatVeg_composition_old()` here and elsewhere
- [ ] Please add R code that calculates `clim1`. I guess `clim1 <- calc_SiteClimate(weatherList = rSOILWAT2::get_WeatherHistory(rSOILWAT2::sw_exampleData), do_C4vars = TRUE)`?